webforj-foundation/src/main/java/com/webforj/optiondialog/MessageDialog.java

```diff
@@ -0,0 +1,374 @@
+    return firstButtonText;
+  }
+
+  /**
+   * Sets the text of the second button.
+   *
+   * @param secondButtonText the text of the second button
+   * @return the message box
+   */
+  public MessageDialog setSecondButtonText(String secondButtonText) {
+    this.secondButtonText = secondButtonText == null ? "" : secondButtonText;
+    return this;
+  }
+
+  /**
+   * Gets the text of the second button.
+   *
+   * @return the text of the second button
+   */
+  public String getSecondButtonText() {
+    return secondButtonText;
+  }
+
+  /**
+   * Sets the text of the third button.
+   *
+   * @param thirdButtonText the text of the third button
+   * @return the message box
+   */
+  public MessageDialog setThirdButtonText(String thirdButtonText) {
+    this.thirdButtonText = thirdButtonText == null ? "" : thirdButtonText;
+    return this;
+  }
+
+  /**
+   * Gets the text of the third button.
+   *
+   * @return the text of the third button
+   */
+  public String getThirdButtonText() {
+    return thirdButtonText;
+  }
+
+  /**
+   * Sets buttons texts.
+   *
+   * @param firstButtonText the text of the first button
+   * @param secondButtonText the text of the second button
+   * @param thirdButtonText the text of the third button
+   *
+   * @return the message box
+   */
+  public MessageDialog setButtonText(String firstButtonText, String secondButtonText,
+      String thirdButtonText) {
+    setFirstButtonText(firstButtonText);
+    setSecondButtonText(secondButtonText);
+    setThirdButtonText(thirdButtonText);
+    return this;
+  }
+
+  /**
+   * Sets buttons texts.
+   *
+   * @param firstButtonText the text of the first button
+   * @param secondButtonText the text of the second button
+   *
+   * @return the message box
+   */
+  public MessageDialog setButtonText(String firstButtonText, String secondButtonText) {
+    setFirstButtonText(firstButtonText);
+    setSecondButtonText(secondButtonText);
+    return this;
+  }
+
+  /**
+   * Sets buttons texts.
+   *
+   * @param firstButtonText the text of the first button
+   *
+   * @return the message box
+   */
+  public MessageDialog setButtonText(String firstButtonText) {
+    setFirstButtonText(firstButtonText);
+    return this;
+  }
+
+  /**
+   * Sets the timeout of the message box.
+   *
+   * @param timeout the timeout of the message box
+   * @return the message box
+   */
+  public MessageDialog setTimeout(int timeout) {
+    if (timeout < 0) {
+      throw new IllegalArgumentException("Timeout must be greater than or equal to 0");
+    }
+
+    this.timeout = timeout;
+    return this;
+  }
+
+  /**
+   * Gets the timeout of the message box.
+   *
+   * @return the timeout of the message box
+   */
+  public int getTimeout() {
+    return timeout;
+  }
+
+  /**
+   * Shows the message box.
+   *
+   * @return the result of the message box
+   */
+  public int show() {
+    return Environment.getCurrent().getWeforjHelper().msgbox(this);
```

Comment:
This return type may be mapped to the ENUM instead of passing the integer directly, so the return type is the enum

Reply:
yes, planning to do this